packages/@lwc/engine-dom/src/apis/create-element.ts

```diff
@@ -150,11 +149,5 @@ export function createElement(
         connectedCallback,
         disconnectedCallback
     );
-    if (!wasComponentUpgraded) {
-        /* eslint-disable-next-line no-console */
-        console.error(
-            `Unexpected tag name "${tagName}". This name is a registered custom element, preventing LWC to upgrade the element.`
-        );
-    }
```

Comment:
This was never actually tested, but I moved this code elsewhere, and there is now a test that covers this.